llvm/lib/Target/AMDGPU/Utils/AMDGPUAsanInstrumentation.cpp

```diff
@@ -270,10 +277,14 @@ void getInterestingMemoryOperands(
     }
     default:
       for (unsigned ArgNo = 0; ArgNo < CI->arg_size(); ArgNo++) {
-        if (!CI->isByValArgument(ArgNo))
-          continue;
-        Type *Ty = CI->getParamByValType(ArgNo);
-        Interesting.emplace_back(I, ArgNo, false, Ty, Align(1));
+        Value *Arg = CI->getArgOperand(ArgNo);
+        if (Arg->getType()->isPointerTy()) {
+          Type *Ty = Arg->getType();
+          Interesting.emplace_back(I, ArgNo, false, Ty, Align(1));
```

Comment:
This handling doesn't make sense. You can't just take the pointer type and use it the same way as the byval type. This needs testing, and explicit byref support